src/components/Checkbox/Checkbox.tsx

```diff
@@ -40,89 +40,96 @@ export default function Checkbox(props: ICheckboxProps) {
   } = props
 
   return (
```

Comment:
I couldn't comment on the function definition line as it's not part of the changes, but I just noticed that you can't pass a ref to the checkbox, which is something that some form libraries rely on and may be useful for other purposes.

As there already other changes on this component, I'd suggest also adding forwardRef to it so we can pass a ref if needed